contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -5,25 +5,33 @@ import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
 import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
 import {IFunctionsCoordinator} from "./interfaces/IFunctionsCoordinator.sol";
 import {IAccessController} from "../../../shared/interfaces/IAccessController.sol";
+import {IConfigurable} from "./interfaces/IConfigurable.sol";
 
-import {RouterBase} from "./RouterBase.sol";
 import {FunctionsSubscriptions} from "./FunctionsSubscriptions.sol";
 import {FunctionsResponse} from "./libraries/FunctionsResponse.sol";
+import {ConfirmedOwner} from "../../../ConfirmedOwner.sol";
 
 import {SafeCast} from "../../../vendor/openzeppelin-solidity/v4.8.0/contracts/utils/SafeCast.sol";
+import {Pausable} from "../../../vendor/openzeppelin-solidity/v4.8.0/contracts/security/Pausable.sol";
 
-contract FunctionsRouter is RouterBase, IFunctionsRouter, FunctionsSubscriptions {
+contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable, ITypeAndVersion, ConfirmedOwner {
   using FunctionsResponse for FunctionsResponse.Commitment;
   using FunctionsResponse for FunctionsResponse.FulfillResult;
 
-  // @inheritdoc ITypeAndVersion
   string public constant override typeAndVersion = "Functions Router v1.0.0";
 
   // We limit return data to a selector plus 4 words. This is to avoid
   // malicious contracts from returning large amounts of data and causing
   // repeated out-of-gas scenarios.
   uint16 public constant MAX_CALLBACK_RETURN_BYTES = 4 + 4 * 32;
 
+  mapping(bytes32 id => address routableContract) private s_route;
+
+  error RouteNotFound(bytes32 id);
+
+  // Use empty bytes to self-identify, since it does not have an id
+  bytes32 private constant ROUTER_ID = bytes32(0);
+
   event RequestStart(
```

Comment:
There is an ask from FE to include "estimated total cost in juels" to RequestStart event.

Reply:
Added!

Reply:
Is there a world where we just want the whole commitment struct emitted here?